src/renderer/rendered_window.rs

```diff
@@ -121,10 +121,15 @@ impl LocatedSurface {
     }
 }
 
+enum PositionOverride {
+    None,
+    Override { top_line: u64, current_scroll: f32 },
+}
+
```

Comment:
I think using `Option` would be more idiomatic and have this be just the struct or maybe even only use a tuple (since variable name should give enough context). You could then do `.unwrap_or((self.current_surface.top_line, self.current_scroll))` (or the struct version). And `is_some`/`is_none`.

Reply:
Yeah, that makes sense.
I'll change it when I can if you dont want to.

Reply:
I'll just do that.